youtube_dl/extractor/pandatv.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    ExtractorError,
+    qualities
+)
+
+class PandaTVIE(InfoExtractor):
+    IE_DESC = '熊猫TV'
+    _VALID_URL = r'http://(?:www\.)?panda\.tv/(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'http://www.panda.tv/10091',
+        'info_dict': {
+            'id': '10091',
+            'title': 're:.+',
+            'uploader': '囚徒',
+            'ext': 'flv',
+            'is_live': True,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        config = self._download_json(
+            'http://www.panda.tv/api_room?roomid=%s' % video_id,
+            video_id
+        )
+
+        data = config['data']
+
+        error_code = config.get('errno', 0)
+        if error_code is not 0:
+            error_desc = 'Server reported error %i' % error_code
+            if isinstance(data, compat_str):
+                error_desc += ': ' + data
+            raise ExtractorError(error_desc, expected=True)
+
+        video_info = data.get('videoinfo')
+
+        # 2 = live, 3 = offline
+        if video_info.get('status') != '2':
+            raise ExtractorError(
+                'Live stream is offline', expected=True)
+
+        title = data['roominfo']['name']
+        uploader = data.get('hostinfo', {}).get('name')
+        plflag = video_info['plflag'].split('_')[1]
+        room_key = video_info['room_key']
+        stream_addr = video_info['stream_addr']
+
+        quality_key = qualities(['HD', 'SD'])
+        suffix = ['_mid', '']
+        formats = []
+        for k, v in stream_addr.items():
+            if v == '1':
+                quality = quality_key(k)
+                if quality >= 0:
+                    formats.append({
+                        'url': 'http://pl%s.live.panda.tv/live_panda/%s%s.flv' % (plflag, room_key, suffix[quality]),
```

Comment:
m3u8 formats available using this template `http://pl-hls%s.live.panda.tv/live_panda/%s%s.m3u8`.